packages/query-engine/source/utils/query-builder.ts

```diff
@@ -0,0 +1,83 @@
+import { Knex, knex } from "knex";
+
+export class QueryBuilder {
+    knexInstance: Knex;
+
+    constructor(knexInstance: Knex) {
+        this.knexInstance = knexInstance;
+    }
+
+    /* TODO: Add the following contructors in if clause of the current
+     * constructor to support constructor overloading. Not required right now
+     * as the current implementation is not using any other constructor.
+     *
+     * constructor(knexInstance: Knex, tableName: string) {
+     *    this.knexInstance = knexInstance.table(tableName);
+     * }
+     *
+     * constructor(knexInstance: Knex, tableName: string, alias: string) {
+     *   this.knexInstance = knexInstance.table(tableName).as(alias);
+     * }
+     *
+     * constructor(config: Knex.Config) {
+     *   this.knexInstance = knex(config);
+     * }
+     */
```

Comment:
Does TypeScript support constructor overloading?

Reply:
TypeScript allows you to declare overloads but you can only have one implementation and that implementation must have a signature that is compatible with all overloads. If that is too messy, we can use the JS pattern with one constructor with switch statements based on args.